io.github.qtox.qTox.json

```diff
@@ -0,0 +1,100 @@
+{
+	"app-id": "io.github.qtox.qTox",
+	"runtime": "org.kde.Platform",
+	"sdk": "org.kde.Sdk",
+	"runtime-version": "5.10",
+	"command": "qtox",
+	"rename-icon": "qtox",
+	"rename-desktop-file": "qtox.desktop",
+	"rename-appdata-file": "qTox.appdata.xml",
+	"finish-args": [
+		"--share=network",
+		"--socket=pulseaudio",
+		"--socket=wayland",
+		"--socket=x11",
+		"--share=ipc",
+		"--filesystem=host",
```

Comment:
Why does it need `host` access? Aren't xdg directories or home enough?

Reply:
To access files from e.g. a usb drive, see also https://github.com/qTox/qTox/issues/5139 for the ongoing discussion.

Reply:
@sudden6 I could set:
"--filesystem=home",
"--filesystem=/media",

Reply:
@robschi Why not using the xdg dirs instead of full home access?

Reply:
@AsciiWolf I don't know, I personally don't store everything in the xdg dirs, but I could always use the override feature I guess:
http://docs.flatpak.org/en/latest/flatpak-command-reference.html?highlight=override#flatpak-override .


So would this be ok?

"--filesystem=xdg-desktop",
"--filesystem=xdg-documents",
"--filesystem=xdg-download",
"--filesystem=xdg-music",
"--filesystem=xdg-pictures",
"--filesystem=xdg-public-share",
"--filesystem=xdg-videos",
"--filesystem=xdg-templates",
"--filesystem=/media",

Reply:
@robschi LGTM, although I am not entirely sure about the xdg-public-share and xdg-templates dirs. Are they really necessary?

Reply:
@AsciiWolf to be honest, I don't really know for what xdg-public-share and xdg-templates are for, so I don't add them, thank you very much for your answer!

Reply:
@robschi 
xdg-public-share -> User's public (Samba) share directory.
xdg-templates -> Directory with templates that can be used when creating new documents.

Reply:
@AsciiWolf thanks for the clarification

Reply:
What is the reason to limit filesystem access?

Personally I would want to be able to send any file, independent of it's storage location.

Reply:
I think the reason for it is that sandboxing aspect.
A app itself should not be able read and write in configs of other apps for example.
And as far as I have understand for selecting files this https://github.com/qTox/qTox/issues/5139 is how it is intended to work.

But it's always possible to overrule the access which is defined in the json file anyhow, just run the flatpak with the following command, and you have "host" access again, if you want:
flatpak run --filesystem=host io.github.qtox.qTox

Reply:
@sudden6 users are able to choose any file they want via the file chooser portal. The portal will then give access to the app for the picked file.

Reply:
@naxuroqa But the application has to support native file chooser dialogs for all its file-related actions. I am not sure if qTox does.

Reply:
@naxuroqa I think we should use the workaround from qTox/flathub#10 for now, so qTox can be added to Flathub before this will be properly fixed upstream.